components/pkg-export-kubernetes/src/manifestjson.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use failure::SyncFailure;
+use handlebars::Handlebars;
+use serde_json::Value;
+
+use export_docker::Result;
+
+use manifest::Manifest;
+
+// Kubernetes manifest template
+const MANIFESTFILE: &'static str = include_str!("../defaults/KubernetesManifest.hbs");
+const BINDFILE: &'static str = include_str!("../defaults/KubernetesBind.hbs");
+
+pub struct ManifestJson {
+    pub main: Value,
+    pub binds: Vec<Value>,
+}
+
+impl ManifestJson {
+    pub fn new(manifest: &Manifest) -> Self {
```

Comment:
What's the advantage of implementing things this way, versus implementing JSON serialization on Manifest itself?

Reply:
As the commit log says:

>     pkg-export-k8s: Add ManifestJson
>
>     Separate out JSON representations of the manifest into its own
>     struct/module. This is to allow users of the API to be able to modify the
>     JSON before writing it to a file.

This was done for the Helm exporter in question.Please refer to chart.rs in this PR, where I am manually generating this struct from the `Manifest`.

These question give me good ideas for my upcoming PR to add docs and tests in k8s exporter crate. :)